nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateTableFetch.java

```diff
@@ -223,19 +237,34 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
             }
             final int numberOfFetches = (partitionSize == 0) ? rowCount : (rowCount / partitionSize) + (rowCount % partitionSize == 0 ? 0 : 1);
 
+            if("null".equals(indexValue)) {
+                // Generate SQL statements to read "pages" of data
+                for (int i = 0; i < numberOfFetches; i++) {
+                    FlowFile sqlFlowFile;
 
-            // Generate SQL statements to read "pages" of data
-            for (int i = 0; i < numberOfFetches; i++) {
-                FlowFile sqlFlowFile;
+                    Integer limit = partitionSize == 0 ? null : partitionSize;
+                    Integer offset = partitionSize == 0 ? null : i * partitionSize;
+                    final String query = dbAdapter.getSelectStatement(tableName, columnNames, whereClause, StringUtils.join(maxValueColumnNameList, ", "), limit, offset);
+                    sqlFlowFile = session.create();
+                    sqlFlowFile = session.write(sqlFlowFile, out -> {
+                        out.write(query.getBytes());
+                    });
+                    session.transfer(sqlFlowFile, REL_SUCCESS);
+                }
+            }else {
+                for (int i = 0; i < numberOfFetches; i++) {
+                    FlowFile sqlFlowFile;
 
-                Integer limit = partitionSize == 0 ? null : partitionSize;
-                Integer offset = partitionSize == 0 ? null : i * partitionSize;
-                final String query = dbAdapter.getSelectStatement(tableName, columnNames, whereClause, StringUtils.join(maxValueColumnNameList, ", "), limit, offset);
-                sqlFlowFile = session.create();
-                sqlFlowFile = session.write(sqlFlowFile, out -> {
-                    out.write(query.getBytes());
-                });
-                session.transfer(sqlFlowFile, REL_SUCCESS);
+                    Integer limit = partitionSize;
+                    whereClause = indexValue + " >= " + limit * i;
```

Comment:
GenerateTableFetch stores max value columns in managed state, so that when it runs again, it fetches only records those have grater max column values.

If we rewrite the `whereClause` here based on only partitionSize, it breaks this behavior when it runs the 2nd time or later.